### PR TITLE
Defer to Kubernetes factory where appropriate

### PR DESCRIPTION
**Pull request by kargakis**
@ironcladlou @deads2k PTAL

### CODE REVIEW

**Comment by deads2k** [comment]
Ick.  This helps, but it doesn't solve the overall factorization problem in `cmdutil.Factory`.

File: pkg/cmd/util/clientcmd/factory.go [hunk 1]
```diff
@@ -18,8 +18,8 @@ import (
 	"github.com/openshift/origin/pkg/client"
 	"github.com/openshift/origin/pkg/cmd/cli/describe"
 	deployapi "github.com/openshift/origin/pkg/deploy/api"
-	deployreaper "github.com/openshift/origin/pkg/deploy/reaper"
-	deploy "github.com/openshift/origin/pkg/deploy/scaler"
+	"github.com/openshift/origin/pkg/deploy/reaper"
```
Comment (deads2k) [inline pkg/cmd/util/clientcmd/factory.go]: I prefer the aliasing for `deployreaper` and `deployscaler`

**Comment by liggitt** [comment]
Double check we weren't overriding implementations for compatibility or bug fix reasons

File: pkg/cmd/util/clientcmd/factory.go (hunk 1 above)
Comment (kargakis) [inline pkg/cmd/util/clientcmd/factory.go]: done

**Comment by kargakis** [comment]
> Ick. This helps, but it doesn't solve the overall factorization problem in cmdutil.Factory.

Can you elaborate on that problem?

**Comment by ironcladlou** [comment]
+1 to removing the fallthrough in the deploy bits. I'm not 100% sure whether `PodSelectorForObject` or `PortsForObject` are equivalent to the old code without diving more into the upstream code.

**Comment by liggitt** [comment]
Also we might want to check if something is an origin type before sending it to the kubernetes factory… not sure how all the factory methods behave with unrecognized types

**Comment by kargakis** [comment]
> +1 to removing the fallthrough in the deploy bits. I'm not 100% sure whether PodSelectorForObject or PortsForObject are equivalent to the old code without diving more into the upstream code.

By old code you mean back when they were `*ForResource`? They are still the same, they just handle objects now instead of querying the server to get them.

@liggitt as long as there are meta on the object, there is no problem.

**Comment by kargakis** [comment]
http://pastebin.test.redhat.com/299288

**Comment by deads2k** [comment]
> Ick. This helps, but it doesn't solve the overall factorization problem in cmdutil.Factory.
> Can you elaborate on that problem?

The upstream `Factory` should be built on a series of a interfaces to make the composition easier.  It's on my list for this/next week.

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**